force-app/main/default/classes/Gifts.cls

```diff
@@ -54,21 +65,12 @@ public with sharing class Gifts {
         return giftsToReturn;
     }
 
-    public void giftsToProcessingStatus() {
+    public void prepareGiftsForProcessing() {
         giftsToUpdate.clear();
-        for (DataImport__c gift : dataImports) {
-            if (isProcessableGift(gift)) {
-                gift.Status__c = BDI_DataImport_API.bdiProcessing;
-                giftsToUpdate.add(gift);
-            }
-        }
+        giftsToProcessingStatus();
```

Comment:
Right now this method does two things - I'd like to see the method split and both be expressed here at the call site for clarity - if(hasPmts && serviceUnavail()) {failGiftsWithPaymentsWithReason('reason'); } else {giftsToProcessingStatus();}